### DISCUSSION TITLE
Form reload after update

### DISCUSSION THREAD

**Discussion by ewald1976** [filamentphp/filament#985] [Help]
Hello,

maybe a silly question but I cannot figure it out. I'd like to show a record_modified timestamp and a record_modified_user field in my form that is updated in the database when i submit an update from an edit form. After the submit the field "record_modified" still has the old value but in the database is already a new value from the update. Is there a way to reload all data in a form or am I missing something?

I use the mutateFormDataBeforeSave method to update those fields.

```
protected function mutateFormDataBeforeSave(array $data): array
{
    $data['Record_Modified'] = Carbon::now()->format('Y-m-d H:i:s.u');
    $data['Record_Modified_User'] = Auth::user()->username;

    return $data;
}
```

Thank you very much in advance.

Kind regards!

**Comment by danharrin** [comment]
Do you have two fields called `Record_Modified` and `Record_Modified_User`

**Reply by ewald1976** [reply]
Yes, there is a "Record_Modified" and a "Record_Modified_User" field in the database.

**Reply by danharrin** [reply]
No, I mean fields in your form.

**Reply by ewald1976** [reply]
Yes, those two fields are also in the form.
```
Forms\Components\TextInput::make('Record_Modified')->disabled(),
Forms\Components\TextInput::make('Record_Modified_User')->disabled(),
```

**Reply by danharrin** [reply]
You should try this instead of `mutateFormDataBeforeSave`:

```php
Forms\Components\TextInput::make('Record_Modified')->dehydrateStateUsing(function ($component, $set) {
    return $set($component, Carbon::now()->format('Y-m-d H:i:s.u'));
})->disabled(),

Forms\Components\TextInput::make('Record_Modified_User')->dehydrateStateUsing(function ($component, $set) {
    return $set($component, Auth::user()->username);
})->disabled(),
```

**Comment by ewald1976** [comment]
That's it! Thank you very much for your help and have a nice holiday.

**Reply by danharrin** [reply]
You too!
